include/gnc/utilities.hpp

```diff
@@ -67,15 +67,32 @@ constexpr void rotate_frame(lin::Vector<T, 4> const &q,
 template <typename T>
 constexpr void rotate_frame(lin::Vector<T, 4> const &q, lin::Vector<T, 3> &v);
 
+/** @fn dcm
+ *  @param[out] DCM Direction cosine matrix.
+ *  @param[in]  x
+ *  @param[in]  y
+ *  Generates a DCM from two reference vector in a particular frame. The DCM
+ *  will convert from the frame x and y are specified in to the new frame.
+ *  The x input vector is normalized to give the first basis vector of the new
+ *  frame. The y vector's projection along x is removed resulting in the second
+ *  basis vector for the new frame. The third is generated assuming a right hand
+ *  coordinate system.
+ *  There is no explicit handling of NaNs built into this functions; however, a
```

Comment:
Add some preconditions that x and y aren't parallel or anti parallel and are non zero.

Reply:
Actually, I just added a tolerance check on the vectors being parallel/anti-parallel and set `DCM` to `NaN`s if the tolerance is violated.